packages/synapse-client/src/util/fetchWithExponentialTimeout.ts

```diff
@@ -25,16 +25,19 @@ export function delay(t: number) {
 const RETRY_STATUS_CODES = [0, 429, 502, 503, 504]
 const MAX_RETRY_STATUS_CODES = [502, 503]
 const MAX_RETRY = 3
+
 /**
- * Fetches data, retrying if the HTTP status code indicates that it could be retried. Contains custom logic for
- * handling errors returned by the Synapse backend.
+ * Fetches data, retrying if the HTTP status code indicates that it could be retried.
+ * To use it in our generated client, this function must be SIDE-EFFECT FREE, so it does not
+ * consume the response body.
+ *
  * @throws SynapseClientError
  */
-export const fetchWithExponentialTimeout = async <TResponse>(
+export const fetchResponseWithExponentialTimeout = async (
```

Comment:
Refactor to create a new method that does not consume the response stream, which caused problems when the generated code also tried to consume it.